CMakeLists.txt

```diff
@@ -20,6 +20,27 @@ project(mgclient VERSION 1.1.0)
 set(mgclient_SOVERSION 2)
 add_definitions(-DMGCLIENT_VERSION="${mgclient_VERSION}")
 
+# Deal with the operating system.
+if(WIN32)
+    message(STATUS "ON WINDOWS BUILD")
+    set(MGCLIENT_ON_WINDOWS TRUE)
+    add_definitions(-DMGCLIENT_ON_WINDOWS=1)
```

Comment:
https://cmake.org/cmake/help/latest/command/add_compile_definitions.html
Is `=1`  necessary? you're only checking with `ifdef`